lms/services/oauth2_token.py

```diff
@@ -59,8 +82,24 @@ def get(self):
                 "We don't have an OAuth 2 token for this user"
             ) from err
 
+    def _token_request(self, token_url, data, auth):
+        response = self._http_service.post(token_url, data=data, auth=auth)
+
+        validated_data = OAuthTokenResponseSchema(response).parse()
+
+        self.save(
+            validated_data["access_token"],
+            validated_data.get("refresh_token"),
+            validated_data.get("expires_in"),  # pylint:disable=no-member
+        )
+
+        return validated_data["access_token"]
```

Comment:
This is split into a separate helper method because it's also going to be called by `refresh_token_request()` in future, see https://github.com/hypothesis/lms/pull/2988